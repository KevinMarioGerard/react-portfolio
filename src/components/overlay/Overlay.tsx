import * as React from 'react';

const Overlay: React.FC = props => (
    <>
        <div className="overlay">
            <div className="overlay__content">{props.children}</div>
        </div>
    </>
);

export default Overlay;
