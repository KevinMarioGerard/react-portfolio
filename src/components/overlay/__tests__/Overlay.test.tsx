import { render } from '@testing-library/react';
import Overlay from '../Overlay';
import * as React from 'react';

describe('Overlay Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Overlay />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <Overlay>
                <div>Overlay Content</div>
            </Overlay>
        );

        expect(container).toMatchSnapshot();
    });
});
