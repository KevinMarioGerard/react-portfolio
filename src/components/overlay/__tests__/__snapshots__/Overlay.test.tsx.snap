// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Component tests should match snapshot 1`] = `
<div>
  <div
    class="overlay"
  >
    <div
      class="overlay__content"
    >
      <div>
        Overlay Content
      </div>
    </div>
  </div>
</div>
`;
